### ISSUE TITLE
`H5FD__init_package` totally unused (it seems to me, and according to clang)

### ISSUE THREAD

**Issue by seanm** [HDFGroup/hdf5#349]
Many files have a function `H5FD__init_package`, usually `static`.  It appears totally unused.  Clang warns:

```
warning: unused function 'H5FD__init_package' [-Wunused-function]
```

Am I blind or is it really unused?

**Comment by qkoziol** [comment]
It's composed as a result of the H5_PACKAGE_INIT_FUNC "glue" macro, from the 'H5_MY_PKG' macro in src/H5FDmodule.h.   It's not a great fit in the way it's currently used, but would be some work to refactor to eliminate. :-/

**Comment by qkoziol** [comment]
This is used in the H5FDmodule.h header's macros

**Comment by jhendersonHDF** [comment]
> This is used in the H5FDmodule.h header's macros

I initially thought this issue might be related because the previously-linked PR does clean up some warnings related to unused H5FD package variables and undefined H5FD routines, but it seems this is separate from that.

**Comment by derobins** [comment]
The reworked VFD initialization routines do not have this problem. These are in develop (1.13.x, eventually 1.14.0). They will probably not be merged down to 1.12 and earlier.
